wic/src/test/java/edu/asu/diging/wic/core/service/impl/ConceptTextServiceTest.java

```diff
@@ -0,0 +1,144 @@
+        
+        Page<ConceptText> dataFromDb = new PageImpl<ConceptText>(dataObj);
+        when(conceptTextDatabaseConnection.findAll(any(Pageable.class)))
+            .thenReturn(dataFromDb);
+        
+        List<ConceptText> objFromDb = conceptTextService.findAll(1, 2);
+        Assert.assertEquals(objFromDb.size(), 2);
+        verify(conceptTextDatabaseConnection).findAll(any(Pageable.class));
+    }
+
+    @Test
+    public void testDeleteText_forSuccess() {
+        conceptTextService.deleteText(1L);
+        verify(conceptTextDatabaseConnection, times(1)).deleteById(eq(1L));
+    }
+    
+    @Test
+    public void testDeleteText_whenIdIsNull() {
+        conceptTextService.deleteText(null);
+        verify(conceptTextDatabaseConnection, never()).deleteById(null);
+    }
+
+    @Test
+    public void testUpdateText() {
+        ConceptText updatedForm = new ConceptText();
+        updatedForm.setText("Updated text");
+        updatedForm.setTitle("Updated title");
+        updatedForm.setAuthor("Updated author");
+        
+        ConceptText oldForm = new ConceptText();
+        oldForm.setText("Old text");
+        oldForm.setTitle("Old title");
+        oldForm.setAuthor("Old Author");
+        oldForm.setModifiedby("Admin");
+        
+        OffsetDateTime dateOnWhichModified = OffsetDateTime.now();
+        oldForm.setModifiedOn(dateOnWhichModified);
+        
+        Optional<ConceptText> oldObjFromDb = Optional.of(oldForm);
+        when(conceptTextDatabaseConnection.findById(1L)).thenReturn(oldObjFromDb);
+        
+        ConceptText updatedSavedObj = conceptTextService.updateText(updatedForm, "Smit", 1L);
+        
+        Assert.assertEquals(updatedSavedObj.getText(), "Updated text");
+        Assert.assertEquals(updatedSavedObj.getTitle(), "Updated title");
+        Assert.assertEquals(updatedSavedObj.getAuthor(), "Updated author");
+        Assert.assertEquals(updatedSavedObj.getModifiedby(), "Smit");
+        
+        verify(conceptTextDatabaseConnection, times(1)).findById(1L);
+        verify(conceptTextDatabaseConnection, times(1)).save(oldForm);
+    }
+
+    @Test
+    public void testGetTextById() {
+        ConceptText dataInDb = new ConceptText();
+        dataInDb.setId(1L);
+        Optional<ConceptText> data = Optional.of(dataInDb);
+        
+        when(conceptTextDatabaseConnection.findById(1L)).thenReturn(data);
+        
+        ConceptText dataFromGetById = conceptTextService.getTextById(1L);
+        
+        Assert.assertEquals(dataInDb, dataFromGetById);
+        verify(conceptTextDatabaseConnection, times(1)).findById(1L);
+    }
+
+    @Test
+    public void testGetTextCount() {
+        
+        when(conceptTextDatabaseConnection.count()).thenReturn(10L);
+        
+        long totalCountInDb = conceptTextService.getTextCount();
+        
+        Assert.assertEquals(10L, totalCountInDb);
+        verify(conceptTextDatabaseConnection,times(1)).count();
```

Comment:
that doesn't really matter. if the total count is found in a different way the outcome can still be correct.